pkg/kv/kvserver/rangefeed/registry.go

```diff
@@ -70,7 +96,10 @@ type registration struct {
 	// Internal.
 	id   int64
 	keys interval.Range
-	buf  chan *roachpb.RangeFeedEvent
+	buf  chan *sharedEvent
+	// releaseAlloc is a callback to notify processor when event is no longer in
+	// use (after processing or when registration stops and discards its queue)
+	releaseAlloc releaseAllocation
```

Comment:
If we made `sharedBudgetAllocation.release()` have a reference to the `feedBudget` and call into it automatically as appropriate, then downstream consumers don't need to have this coupling to whatever created the allocation. They can just call `release()` and get on with their day.